Source/Events.Handling/Builder/EventHandlerBuilder.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using Dolittle.SDK.Events.Handling.Internal;
+using Dolittle.SDK.Events.Processing;
+using Microsoft.Extensions.Logging;
+
+namespace Dolittle.SDK.Events.Handling.Builder
+{
+    /// <summary>
+    /// Represents a building event handlers.
+    /// </summary>
+    public class EventHandlerBuilder
+    {
+        readonly EventHandlerId _eventHandlerId;
+        readonly ILoggerFactory _loggerFactory;
+        readonly ILogger _logger;
+        EventHandlerMethodsBuilder _methodsBuilder;
+
+        ScopeId _scopeId = ScopeId.Default;
+
+        bool _partitioned = true;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventHandlerBuilder"/> class.
+        /// </summary>
+        /// <param name="eventHandlerId">The <see cref="EventHandlerId" />.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory" />.</param>
+        public EventHandlerBuilder(EventHandlerId eventHandlerId, ILoggerFactory loggerFactory)
+        {
+            _eventHandlerId = eventHandlerId;
+            _loggerFactory = loggerFactory;
+            _logger = loggerFactory.CreateLogger<EventHandlerBuilder>();
+        }
+
+        /// <summary>
+        /// Defines the event handler to be partitioned - this is default for an event handler.
+        /// </summary>
+        /// <returns>The builder for continuation.</returns>
+        public EventHandlerBuilder Partitioned()
+        {
+            _partitioned = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Defines the event handler to be unpartitioned. By default it will be partitioned.
+        /// </summary>
+        /// <returns>The builder for continuation.</returns>
+        public EventHandlerBuilder Unpartitioned()
+        {
+            _partitioned = false;
+            return this;
+        }
+
+        /// <summary>
+        /// Defines the event handler to operate on a specific <see cref="_scopeId" />.
+        /// </summary>
+        /// <param name="scopeId">The <see cref="_scopeId" />.</param>
+        /// <returns>The builder for continuation.</returns>
+        public EventHandlerBuilder InScope(ScopeId scopeId)
+        {
+            _scopeId = scopeId;
+            return this;
+        }
+
+        /// <summary>
+        /// Build event handler methods.
+        /// </summary>
+        /// <param name="callback">The callback for building event handler methods.</param>
+        public void WithMethods(Action<EventHandlerMethodsBuilder> callback)
+        {
+            _methodsBuilder = new EventHandlerMethodsBuilder(_eventHandlerId, _loggerFactory.CreateLogger<EventHandlerMethodsBuilder>());
+            callback(_methodsBuilder);
+        }
```

Comment:
Personally - I think this extra layer of builder callback is not very pleasing on the eye. And I don't see any reason why we _need_ it?

Reply:
This was done to be able to create multiple event handler methods for an event handler. I'm open to suggestions on how to do that in a less layered way

Reply:
We will look at this compared to the guide and TS SDK later.